### PR TITLE
MLO-46: Expose apps config to users

### PR DESCRIPTION
**Pull request by YevheniiSemendiak**
We don't need it in CLI, only in SDK at the moment.

### CODE REVIEW

File: apolo-sdk/src/apolo_sdk/_server_cfg.py [hunk 1]
```diff
@@ -82,6 +82,12 @@ def key(self) -> Key:
         )
 
 
+@rewrite_module
+@dataclass(frozen=True)
+class AppsConfig:
+    apps_hostname_templates: Sequence[str] = ()
```
Comment (asvetlov) [inline apolo-sdk/src/apolo_sdk/_server_cfg.py:88]: I'm curious what is the meaning of this list?
What is the format for the list item?

File: apolo-sdk/src/apolo_sdk/_server_cfg.py [hunk 2]
```diff
@@ -179,6 +186,14 @@ def _parse_cluster_config(payload: Dict[str, Any]) -> Cluster:
         orgs = ["NO_ORG"]
     else:
         orgs = [org if org is not None else "NO_ORG" for org in orgs]
+
+    apps_payload = payload.get("apps", {})
+    apps_config = AppsConfig()
+    if apps_payload:
+        apps_config = AppsConfig(
+            apps_hostname_templates=apps_payload.get("apps_hostname_templates", [])
+        )
```
Comment (asvetlov) [inline apolo-sdk/src/apolo_sdk/_server_cfg.py:195]: Let's avoid calling `AppsConfig` constructor twice.
```suggestion
    if apps_payload:
        apps_config = AppsConfig(
            apps_hostname_templates=apps_payload.get("apps_hostname_templates", [])
        )
    else:
        apps_config = AppsConfig()
```

**Review by asvetlov — COMMENTED** [review]
Should we display this config in commands like `apolo config show`?

Also, the documentation should be updated to reflect SDK changes.

File: apolo-sdk/src/apolo_sdk/_server_cfg.py (hunk 1 above)
Comment (asvetlov) [inline apolo-sdk/src/apolo_sdk/_server_cfg.py:88]: Could we use a shorter name, `apps_hostname_templates` is too clumsy.

Comment (YevheniiSemendiak) [reply apolo-sdk/src/apolo_sdk/_server_cfg.py:88]: This is somewhat similar to `job_hostname_template`. This is a string with a placeholder `"{app_name}.apps.novoserve.org.neu.ro."`.  Also, Ivan mentioned, we might have a list of domain names for a single cluster in a future, therefore, it's a list of such placeholders.

End customers will not use it directly at the moment, therefore, I didn't add it to CLI itself.
We are going to use it in our apps deployment workflow, see https://linear.app/apolo/project/apps-support-various-base-domains-3e84552057e0/overview

File: apolo-sdk/src/apolo_sdk/_server_cfg.py (hunk 2 above)
Comment (YevheniiSemendiak) [reply apolo-sdk/src/apolo_sdk/_server_cfg.py]: applied, thank you!

File: apolo-sdk/src/apolo_sdk/_server_cfg.py (hunk 1 above)
Comment (asvetlov) [reply apolo-sdk/src/apolo_sdk/_server_cfg.py]: ok

**Review by asvetlov — COMMENTED** [review]
Along with documentation update please create `./CHANGELOG.D/3142.feature` file

**Comment by YevheniiSemendiak** [comment]
@asvetlov added docs, changelog. Please, take a glance once more

**Review by asvetlov — APPROVED** [review]
All good, thanks!

**Pull request merged**